Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -6,16 +6,60 @@
 
 #include <algorithm>
 #include <cstddef>
+#include <optional>
 #include <sstream>
 #include <string>
 #include <vector>
 
+#include <expr.h>
+
 #include "Common/CommonTypes.h"
 #include "Common/DebugInterface.h"
 #include "Common/Logging/Log.h"
 #include "Core/Core.h"
 #include "Core/PowerPC/JitInterface.h"
 #include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+struct ExprDeleter
+{
+  void operator()(expr* expression) const { expr_destroy(expression, nullptr); }
+};
+
+using ExprPointer = std::unique_ptr<expr, ExprDeleter>;
+
+static std::optional<int> ParseGPR(const char* name)
+{
+  if (name[0] == 'r' && isdigit(name[1]))
+  {
+    char* end = nullptr;
+    int index = strtol(&name[1], &end, 10);
+    if (index < 32 && *end == '\0')
+      return index;
+  }
+  return {};
+}
+
+static bool EvaluateExpression(const std::string& expression_string)
+{
+  static expr_var_list vars = {0};
+
+  ExprPointer expression{
+      expr_create(expression_string.c_str(), expression_string.length(), &vars, nullptr)};
+  if (!expression)
+    return false;
+
+  for (auto* v = vars.head; v != nullptr; v = v->next)
+  {
+    auto index = ParseGPR(v->name);
+    if (index)
+      v->value = GPR(index.value());
```

Comment:
We typically use `std::optional<T>::operator*` if the optional has already been checked to see if it contains a value:
 
```suggestion
      v->value = GPR(*index);
```